CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLHarvester.cc

```diff
@@ -0,0 +1,160 @@
+  PPSTimingCalibration::TimingMap calib_time;
+
+  iGetter.cd();
+  iGetter.setCurrentFolder(dqmDir_);
+
+  // compute the fit parameters for all monitored channels
+  TimingCalibrationHistograms hists;
+  std::string ch_name;
+  for (const auto& detid : detids_) {
+    detid.channelName(ch_name);
+    const auto chid = detid.rawId();
+    const PPSTimingCalibration::Key key{
+        (int)detid.arm(), (int)detid.station(), (int)detid.plane(), (int)detid.channel()};
+    hists.leadingTime[chid] = iGetter.get("t_" + ch_name);
+    if (hists.leadingTime[chid] == nullptr) {
+      edm::LogInfo("PPSTimingCalibrationPCLHarvester:dqmEndJob")
+          << "Failed to retrieve leading time monitor for channel (" << detid << ").";
+      continue;
+    }
+    hists.toT[chid] = iGetter.get("tot_" + ch_name);
+    if (hists.toT[chid] == nullptr) {
+      edm::LogInfo("PPSTimingCalibrationPCLHarvester:dqmEndJob")
+          << "Failed to retrieve time over threshold monitor for channel (" << detid << ").";
+      continue;
+    }
+    hists.leadingTimeVsToT[chid] = iGetter.get("tvstot_" + ch_name);
+    if (hists.leadingTimeVsToT[chid] == nullptr) {
+      edm::LogInfo("PPSTimingCalibrationPCLHarvester:dqmEndJob")
+          << "Failed to retrieve leading time vs. time over threshold monitor for channel (" << detid << ").";
+      continue;
+    }
+    if (min_entries_ > 0 && hists.leadingTimeVsToT[chid]->getEntries() < min_entries_) {
+      edm::LogWarning("PPSTimingCalibrationPCLHarvester:dqmEndJob")
+          << "Not enough entries for channel (" << detid << "): " << hists.leadingTimeVsToT[chid]->getEntries() << " < "
+          << min_entries_ << ". Skipping calibration.";
+      continue;
+    }
+    const double upper_tot_range = hists.toT[chid]->getMean() + 2.5;
+    auto prof = hists.leadingTimeVsToT[chid]->getTH2D()->ProfileX("_prof_x", 1, -1);
+    interp_.SetParameters(hists.leadingTime[chid]->getRMS(),
+                          hists.toT[chid]->getMean(),
+                          0.8,
+                          hists.leadingTime[chid]->getMean() - hists.leadingTime[chid]->getRMS());
+    const auto& res = prof->Fit(&interp_, "B+", "", 10.4, upper_tot_range);
+    if ((bool)res) {
+      calib_params[key] = {
+          interp_.GetParameter(0), interp_.GetParameter(1), interp_.GetParameter(2), interp_.GetParameter(3)};
+      calib_time[key] = std::make_pair(0.1, 0.);  // hardcoded resolution/offset placeholder for the time being
+      // can possibly do something with interp_.GetChiSquare() in the near future
+    } else
+      edm::LogWarning("PPSTimingCalibrationPCLHarvester:dqmEndJob")
+          << "Fit did not converge for channel (" << detid << ").";
+    delete prof;
+  }
+
+  // fill the DB object record
+  PPSTimingCalibration calib(formula_, calib_params, calib_time);
+
+  // write the object
+  edm::Service<cond::service::PoolDBOutputService> poolDbService;
+  if (!poolDbService.isAvailable())
```

Comment:
assuming that this cannot "disappear" during the job, can you move this check to the constructor or beginJob ?
it seems it would be better to let the job fail as soon as possible, rather than at the end.